automata/fa/nfa.py

```diff
@@ -510,6 +518,39 @@ def intersection(self, other):
             final_states=new_final_states
         )
 
+    def shuffle_product(self, other):
+        """
+        Given two NFAs, M1 and M2, which accept the languages
+        L1 and L2 respectively, returns an NFA which accepts
+        the shuffle of L1 and L2.
+        """
+
```

Comment:
The other operations that involve two NFAs (like `intersection` and `union`) only do this in their dunder methods. Happy to add it here but will then change the others for consistency.

Reply:
@eliotwrobson I mean, it probably makes sense to keep those checks located in the dunder methods for those operations. But since there is no dunder equivalent for `shuffle_product`, I'm honestly not too concerned with consistency in this case.

Reply:
Sounds good, will add it in